memory/src/main/java/org/opensearch/ml/memory/index/ConversationMetaIndex.java

```diff
@@ -336,6 +349,27 @@ public void searchConversations(SearchRequest request, ActionListener<SearchResp
         }
     }
 
+    /**
+     * Update conversations in the index
+     * @param updateRequest original update request
+     * @param listener receives the update response for the wrapped query
+     */
+    public void updateConversation(UpdateRequest updateRequest, ActionListener<UpdateResponse> listener) {
+        if (!clusterService.state().metadata().hasIndex(META_INDEX_NAME)) {
+            listener
+                .onFailure(
+                    new IndexNotFoundException("cannot update conversation since the conversation index does not exist", META_INDEX_NAME)
+                );
+            return;
+        }
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            client.update(updateRequest, listener);
```

Comment:
missed `context::restore` ?

Reply:
Sure added the restore back. Most likely it's not needed here because no further context is needed in the ActionLisenter in the current use cases. But to be consistent it's still better to restore.